internal/characters/traveler/common/hydro/skill.go

```diff
@@ -0,0 +1,334 @@
+	skillShortHoldFrames[1][action.ActionDash] = 74    // Short Hold E -> D
+	skillShortHoldFrames[1][action.ActionJump] = 73    // Short Hold E -> J
+	skillShortHoldFrames[1][action.ActionWalk] = 80    // Short Hold E -> Walk
+}
+
+func (c *char) skillPress(hitmark, spiritHitmark, cdStart int, skillFrames [][]int) (action.Info, error) {
+	c.torrentSurge(hitmark, spiritHitmark)
+	c.SetCDWithDelay(action.ActionSkill, 10*60, cdStart)
+
+	if c.Base.Cons >= 4 {
+		c.c4()
+	}
+
+	return action.Info{
+		Frames:          frames.NewAbilFunc(skillFrames[c.gender]),
+		AnimationLength: skillFrames[c.gender][action.InvalidAction],
+		CanQueueAfter:   skillFrames[c.gender][action.ActionDash], // earliest cancel
+		State:           action.SkillState,
+		OnRemoved:       func(next action.AnimationState) { c.c4Remove() },
+	}, nil
+}
+
+func (c *char) skillShortHold(travel int) (action.Info, error) {
+	aiHold := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Dewdrop (Hold)",
+		AttackTag:  attacks.AttackTagElementalArtHold,
+		ICDTag:     attacks.ICDTagTravelerDewdrop,
+		ICDGroup:   attacks.ICDGroupTravelerDewdrop,
+		StrikeType: attacks.StrikeTypeSlash,
+		Element:    attributes.Hydro,
+		Durability: 25,
+		Mult:       skillDewdrop[c.TalentLvlSkill()],
+	}
+
+	if c.Base.Cons >= 4 {
+		c.c4()
+	}
+
+	c.QueueCharTask(func() {
+		c.skillLosingHP(&aiHold)
+		c.Core.QueueAttack(
+			aiHold,
+			combat.NewBoxHit(c.Core.Combat.Player(), c.Core.Combat.PrimaryTarget(), geometry.Point{Y: -0.4}, 0.3, 1.3),
+			0,
+			1,
+			c.makeA1CB(),
+			c.makeC4CB(),
+		)
+	}, skillShortHoldFirstDewdropRelease+travel)
+
+	c.torrentSurge(skillShortHoldTorrentSurgeHitmark, skillShortHoldSpiritbreathThornHitmark)
+	c.SetCDWithDelay(action.ActionSkill, 10*60, skillShortHoldCdStart)
+
+	return action.Info{
+		Frames:          frames.NewAbilFunc(skillShortHoldFrames[c.gender]),
+		AnimationLength: skillShortHoldFrames[c.gender][action.InvalidAction],
+		CanQueueAfter:   skillShortHoldFrames[c.gender][action.ActionJump], // earliest cancel
+		State:           action.SkillState,
+		OnRemoved:       func(next action.AnimationState) { c.c4Remove() },
+	}, nil
+}
+
+func (c *char) skillHold(travel, holdTicks int) (action.Info, error) {
+	aiHold := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Dewdrop (Hold)",
+		AttackTag:  attacks.AttackTagElementalArtHold,
+		ICDTag:     attacks.ICDTagTravelerDewdrop,
+		ICDGroup:   attacks.ICDGroupTravelerDewdrop,
+		StrikeType: attacks.StrikeTypeSlash,
+		Element:    attributes.Hydro,
+		Durability: 25,
+		Mult:       skillDewdrop[c.TalentLvlSkill()],
+	}
+
+	if c.Base.Cons >= 4 {
+		c.c4()
+	}
+
+	extend := 15 * (holdTicks - 1)
+	for i := 0; i <= extend; i += 15 {
+		c.QueueCharTask(func() {
+			c.skillLosingHP(&aiHold)
+			c.Core.QueueAttack(
+				aiHold,
+				combat.NewBoxHit(c.Core.Combat.Player(), c.Core.Combat.PrimaryTarget(), geometry.Point{Y: -0.4}, 0.3, 1.3),
+				0,
+				1,
+				c.makeA1CB(),
```

Comment:
should be moved outside of the loop so that the callback is shared and count works as intended